ddtrace/appsec/processor.py

```
@@ -0,0 +1,119 @@
+DEFAULT_RULES = os.path.join(ROOT_DIR, "rules.json")
+
+log = get_logger(__name__)
+
+
+@attr.s(eq=False)
+class AppSecProcessor(object):
+
+    tracer = attr.ib(type=ddtrace.Tracer, default=ddtrace.tracer)
+    rules = attr.ib(type=str, factory=lambda: get_env("appsec", "rules", default=DEFAULT_RULES))
+    _ddwaf = attr.ib(type=DDWaf, default=None)
+
+    enabled = False  # type: ClassVar[bool]
+    _instance = None  # type: ClassVar[Optional[AppSecProcessor]]
+
+    def __attrs_post_init__(self):
+        # type: () -> None
+        if self._ddwaf is None:
+            try:
+                with open(self.rules, "r") as f:
+                    rules = json.load(f)
+            except EnvironmentError as err:
+                # DDAS-0001-03
+                if err.errno == errno.ENOENT:
+                    log.error("AppSec could not read the rule file %s. Reason: file does not exist", self.rules)
+                else:
+                    # DDAS-0001-03 - TODO: try to log reasons
+                    log.error("AppSec could not read the rule file %s.", self.rules)
+                raise
+            except json.decoder.JSONDecodeError:
+                # DDAS-0001-03
+                log.error("AppSec could not read the rule file %s. Reason: invalid JSON file", self.rules)
+                raise
+            except Exception:
+                # DDAS-0001-03 - TODO: try to log reasons
+                log.error("AppSec could not read the rule file %s.", self.rules)
+                raise
+            try:
+                self._ddwaf = DDWaf(rules)
+            except ValueError:
+                # Partial of DDAS-0005-00
+                log.warning("WAF initialization failed")
+                raise
+
+    @classmethod
+    def enable(cls, *args, **kwargs):
+        # type: (Any, Any) -> None
+        if cls._instance is not None:
+            return
+
+        try:
+            processor = cls(*args, **kwargs)
```

Should this fail loudly? If appsec is enabled and it fails to load then it should be a hard error imo

---

Please check https://github.com/DataDog/dd-trace-py/pull/3175/files -> we don't want it to fail loudly for now (private alpha), I will align with all AppSec tracers for the next round